src/api/venue.ts

```diff
@@ -34,3 +44,71 @@ export const setVenueLiveStatus = async ({
     })
     .finally(onFinish);
 };
+
+// TODO: return an array of all venueId's we find on the full way up to the sovereignVenue
+export const fetchSovereignVenueId = async (
+  venueId: string,
+  previouslyCheckedVenueIds: readonly string[] = []
+): Promise<string> => {
+  const venue = await fetchVenue(venueId);
+
+  if (!venue) throw new Error(`The '${venueId}' venue doesn't exist`);
+
+  if (!venue.parentId) return venue.id;
+
+  if (previouslyCheckedVenueIds.includes(venueId))
+    throw new Error(
+      `Circular reference detected. '${venueId}' has already been checked`
+    );
+
+  return fetchSovereignVenueId(venue.parentId, [
+    ...previouslyCheckedVenueIds,
+    venueId,
+  ]);
```

Comment:
[note] This should be all that's needed to resolve the potential 'infinite recursion' edgecase bug described in https://github.com/sparkletown/internal-sparkle-issues/issues/699